logic/tool/TypeHinter.java

```diff
@@ -221,12 +214,22 @@ private void addInferredSubLabels(TypeVariable variable, Set<Label> hints, Map<L
     }
 
     private void addInferredIsaLabels(ThingVariable variable, Set<Label> hints, Map<Label, TypeVariable> labelMap) {
+        removeAbstractHints(hints);
```

Comment:
It might have been nice to have done this in `retrieveVariableHints`, since that is whence the type hints come. But that method doesn't have the context of whether a variable is Thing or Type, so it has to be filtered later. Putting it here seems the neatest place, since this is exactly then a Thing Variable is assigned its Hint Labels.

Reply:
i think it'll be fine if you inline this function and just remove the assertion, since we should never have invalid labels in the first place...
`hints.removeIf(label -> conceptMgr.getType(label.scopedName()).isAbstract())`

Reply:
Alright, made this one line.